### ISSUE TITLE
Long colnames get truncated

### ISSUE THREAD

**Issue by FelixTheStudent** [Bioconductor/HDF5Array#24]
Writing out a sparse Matrix seems to work fine (no warnings). However, once I read it back it, colnames get cropped so that they are not longer than 11 characters (in the minimal example below at least; I have had another case where colnames were not longer than 10).

I use HDF5Array 1.12.3.

```{r}
library(Matrix)
library(HDF5Array)

# create example matrix with increasingly long colnames:
mat <- as(matrix(rep(0, 900), nrow=30), "dgCMatrix")
mat[sample(1:30, 3), sample(1:30, 3)] <- 1
colnames(mat) <- sapply(1:30, function(i) paste(rep("x", i), collapse = ""))
# write out:
HDF5Array::writeTENxMatrix(mat,
                filepath=file.path("~", "tmp_file.hdf5"),
                group="tmp", level=NULL, verbose=F)
x <- HDF5Array::TENxMatrix("~/tmp_file.hdf5", group="tmp")
max(stringr::str_length(colnames(mat)))
# 30
max(stringr::str_length(colnames(x)))
# 11
```

Any ideas?

**Comment by hpages** [comment]
Thanks Felix. Should be fixed in HDF5Array 1.15.7 (see commit 0e487076). Will port the fix to BioC 3.10 later today.

H.

**Comment by FelixTheStudent** [comment]
Excellent, thanks. Will tell my admin to update bioconductor.

Cheers,

Felix



Hervé Pagès <notifications@github.com> schrieb am Fr., 28. Feb. 2020, 12:06:

**Comment by hpages** [comment]
Sounds good. I've ported the fix to BioC 3.10. It's in HDF5Array 1.14.3 but please allow 24h for the package to pass the builds and become available via `BiocManager::install()`.

Note that admins should update Bioconductor on a regular basis regardless (dozens of packages get fixed every month in release).

Cheers,
H.
